RecoTracker/LSTCore/src/alpaka/Event.dev.cc

```diff
@@ -194,11 +193,15 @@ void Event::addPixelSegmentToEvent(std::vector<unsigned int> const& hitIndices0,
     *nTotalMDs_buf_h.data() += n_max_pixel_md_per_modules;
     unsigned int nTotalMDs = *nTotalMDs_buf_h.data();
 
-    mdsInGPU_.emplace();
-    miniDoubletsBuffers_.emplace(nTotalMDs, nLowerModules_, devAcc_, queue_);
-    mdsInGPU_->setData(*miniDoubletsBuffers_);
+    std::array<int, 2> const mds_sizes{{static_cast<int>(nTotalMDs), static_cast<int>(nLowerModules_ + 1)}};
+    miniDoubletsDC_.emplace(mds_sizes, queue_);
 
-    alpaka::memcpy(queue_, miniDoubletsBuffers_->nMemoryLocations_buf, nTotalMDs_buf_h);
+    auto nMDs_view =
+        alpaka::createView(devAcc_, miniDoubletsDC_->view<MiniDoubletsOccupancySoA>().nMDs(), nLowerModules_ + 1);
```

Comment:
```suggestion
        alpaka::createView(devAcc_, miniDoubletsDC_->view<MiniDoubletsOccupancySoA>().nMDs(), mds_sizes[1]);
```
or later where the array is not available `miniDoubletsDC_->sizes()[1]` ... or if the lower level layout is available `view.metadata().size()`

Reply:
Oh yeah, that's better